PyInstaller/building/imphook.py

```diff
@@ -12,18 +12,449 @@
 Code related to processing of import hooks.
 """
 
-import glob
+import glob, weakref
 import os.path
 
 from .. import log as logging
-from .utils import format_binaries_and_datas
-from ..compat import expand_path
-from ..compat import importlib_load_source
+from ..compat import expand_path, importlib_load_source, FileNotFoundError
 from .imphookapi import PostGraphAPI
+from .utils import format_binaries_and_datas
 
 logger = logging.getLogger(__name__)
 
 
+class ModuleHookCache(dict):
```

Comment:
> Inheriting from the built-in collection types (`dict`, `list`, and `set`) is discouraged in Python 3.

Guido, _what have you done._

Thanks for the `stackoverflow`ish high-level synopsis. I didn't know _any_ of that, suggesting I should rescind my Pythonista membership. If I could give you upvote karma, you'd be chumming it up with [Alex Martelli](http://stackoverflow.com/users/95810/alex-martelli).

> `collections.UserDict`

Where have you been my entire Python development life? It shall be so.